incubator/hnc/pkg/kubectl/tree.go

```diff
@@ -32,19 +33,35 @@ var (
 )
 
 var treeCmd = &cobra.Command{
-	Use:   "tree",
-	Short: "Displays the hierarchy tree rooted at the given namespace",
-	Args:  cobra.ExactArgs(1),
+	Use:     "tree",
+	Short:   "Displays the hierarchy tree rooted at the given namespaces separated by space and if no namespace is provided then throws an error, and in case of --all-namespace/-A flag shows tree for all root namespaces",
+	Aliases: []string{"tree"},
 	Run: func(cmd *cobra.Command, args []string) {
-		nnm := args[0]
+		flags := cmd.Flags()
 		footnotesByMsg = map[string]int{}
 		footnotes = []string{}
-		hier := getHierarchy(nnm)
-		fmt.Println(nameAndFootnotes(hier))
-		printSubtree("", hier)
-
+		var nsList []string
+		defaultList := len(args) == 0
+		nsList = args
+		if flags.Changed("all-namespaces") {
+			nsList = getAllNamespaces()
+		}
+		if defaultList && !flags.Changed("all-namespaces") {
+			fmt.Printf("Error: requires some argument or flag but provided none\n")
```

Comment:
You should give more guidance than this. Maybe "you must specify the root of the tree(s) to display or else specify --all-namespaces"?